src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestRespondentReviewPreparer.java

```diff
@@ -71,12 +81,13 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
 
         asylumCase.write(SEND_DIRECTION_PARTIES, Parties.RESPONDENT);
 
-        asylumCase.write(SEND_DIRECTION_DATE_DUE,
-            dateProvider
-                .now()
-                .plusDays(requestRespondentReviewDueInDays)
-                .toString()
-        );
+        int dueInDays = HandlerUtils.isAcceleratedDetainedAppeal(asylumCase)
+                ? requestRespondentReviewDueInDaysForAda
+                : requestRespondentReviewDueInDays;
+        LocalDate dueDate = HandlerUtils.isAcceleratedDetainedAppeal(asylumCase)
+                ? dueDateService.calculateDueDate(dateProvider.now().atStartOfDay(ZoneOffset.UTC), dueInDays).toLocalDate()
+                : dateProvider.now().plusDays(dueInDays);
```

Comment:
You can actually use the "requestRespondentEvidenceDueInDaysAda" and "requestRespondentEvidenceDueInDays" directly in calculating the dueDate since you are checking for ADA again. So you can get rid of the variable "dueInDays"